deps: @metamask/network-controller v15

legobeat opened the pull request:
## **Description**

Bump dependencies:
- Complete upgrade of `@metamask/network-controller`@`^14.0.0`->`^15.1.0`
  - Direct dependency bumped in #21668 
  - [CHANGELOG](https://github.com/MetaMask/core/blob/main/packages/network-controller/CHANGELOG.md) 
  - `@metamask/gas-fee-controller`@`^6.0.1`->`^10.0.0` to satisfy peerDependency
    - [CHANGELOG](https://github.com/MetaMask/core/blob/main/packages/gas-fee-controller/CHANGELOG.md) 
  - `@metamask/selected-network-controller`@`^2.0.0`->`^3.1.0` to satisfy peerDependency
    - [CHANGELOG](https://github.com/MetaMask/core/blob/main/packages/selected-network-controller/CHANGELOG.md) 
  - `@metamask/smart-transactions-controller`@`^4.0.0`->`^6.2.0` to satisfy peerDependency
    - [CHANGELOG](https://github.com/MetaMask/smart-transactions-controller/blob/main/CHANGELOG.md) 

Prior to this change, `@metamask/network-controller` is currently included as multiple independent versions:
- `^10.3.0` from `@metamask/gas-fee-controller` and `@metamask/smart-transactions-controller`
- `^13.0.0` from `@metamask/selected-network-controller`
- `^15.0.0` from multiple other packages, and as direct dependency

This consolidates on the single version `15.1.0`. This may make for a notable reduction in bundle size.

## **Related issues**

- #21346 

#### Blocked by

- [x] https://github.com/MetaMask/eth-json-rpc-middleware/issues/262
  - [x] https://github.com/MetaMask/eth-json-rpc-middleware/pull/263

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

legobeat committed:
deps: @metamask/network-controller@^14.0.0->^15.1.0; @metamask/selected-network-controller@^2.0.0->^3.1.0; @metamask/smart-transactions-controller@^4.0.0->^6.0.0

legobeat committed:
deps: @metamask/gas-fee-controller@^6.0.1->^10.0.0

legobeat committed:
add getNetworkClientById option to SmartTransactionsController constructor

legobeat committed:
chore: set allowedActions for GasFeeController messenger wrapper

legobeat committed:
deps: json-rpc-engine@^6.1.0->@metamask/json-rpc-engine^7.2.0

legobeat committed:
use destroy function on json-rpc-engine

legobeat committed:
deps: @metamask/eth-json-rpc-middleware@^12.0.0->^12.0.1

jiexi commented:
`@metamask/network-controller` bump to 15.1.0 completed
https://github.com/MetaMask/metamask-extension/pull/21668

legobeat committed:
deps: @metamask/smart-transactions-controller@^6.0.0->^6.2.0

legobeat committed:
deps: @metamask/network-controller@^15.1.0->^15.2.0

jiexi commented on app/scripts/metamask-controller.js:
```diff
@@ -1505,6 +1509,9 @@ export default class MetamaskController extends EventEmitter {
     );
     this.smartTransactionsController = new SmartTransactionsController(
       {
+        getNetworkClientById: this.networkController.getNetworkClientById.bind(
```
remove `getNetwork` that is being passed in a few lines below this

legobeat committed:
remove getNetwork parameter from SmartTransactionsController constructor call

PR feedback from jiexi <jiexiluan@gmail.com>
  - https://github.com/MetaMask/metamask-extension/pull/21529/files#r1387194841

legobeat committed:
update lavamoat policies

jiexi commented:
Transparency: I'm working on this on my own branch here https://github.com/MetaMask/metamask-extension/pull/21832
